engine/src/main/java/org/destinationsol/menu/MainMenuScreen.java

```diff
@@ -29,93 +28,82 @@
 
-        tutorialControl = new SolUiControl(buttonWidth, buttonHeight, UiDrawer.positions.get("bottom"), 0, calculateButtonOffsetFromBottom(3), true, Input.Keys.T);
-        tutorialControl.setDisplayName("Tutorial");
-        controls.add(tutorialControl);
+        buttonList.addElement(new UiTextButton().setDisplayName("Tutorial")
+                                                .setTriggerKey(Input.Keys.T)
+                                                .enableSound()
+                                                .setOnReleaseAction(() -> solApplication.play(true, "Imperial Small", true)));
 
-        newGameControl = new SolUiControl(buttonWidth, buttonHeight, UiDrawer.positions.get("bottom"), 0, calculateButtonOffsetFromBottom(2), true, gameOptions.getKeyShoot());
-        newGameControl.setDisplayName("Play Game");
-        controls.add(newGameControl);
+        buttonList.addElement(new UiTextButton().setDisplayName("Play Game")
+                                                .setTriggerKey(gameOptions.getKeyShoot())
+                                                .enableSound()
+                                                .setOnReleaseAction(() -> inputManager.changeScreen(screens.newGameScreen)));
 
         // TODO: Temporarily showing on mobile as well. Fix!
         // optionsControl = new SolUiControl(isMobile ? null : menuLayout.buttonRect(-1, 3), true, Input.Keys.O);
-        optionsControl = new SolUiControl(buttonWidth, buttonHeight, UiDrawer.positions.get("bottom"), 0, calculateButtonOffsetFromBottom(1), true, Input.Keys.O);
-        optionsControl.setDisplayName("Options");
-        controls.add(optionsControl);
-
-        exitControl = new SolUiControl(buttonWidth, buttonHeight, UiDrawer.positions.get("bottom"), 0, calculateButtonOffsetFromBottom(0), true, gameOptions.getKeyEscape());
-        exitControl.setDisplayName("Exit");
-        controls.add(exitControl);
-
-        creditsControl = new SolUiControl(MenuLayout.bottomRightFloatingButton(displayDimensions), true, Input.Keys.C);
-        creditsControl.setDisplayName("Credits");
-        controls.add(creditsControl);
+        buttonList.addElement(new UiTextButton().setDisplayName("Options")
+                                                .setTriggerKey(Input.Keys.O)
+                                                .enableSound()
+                                                .setOnReleaseAction(() -> inputManager.changeScreen(screens.optionsScreen)));
+
+        buttonList.addElement(new UiTextButton().setDisplayName("Exit")
+                                                .setTriggerKey(gameOptions.getKeyEscape())
+                                                .enableSound()
+                                                .setOnReleaseAction(() ->   {
+                                                                                // Save the settings on exit, but not on mobile as settings don't exist there.
```

Comment:
This almost looks like **too much** of indentation

Reply:
Whaaaaaat? I can't hear you from all the way over there 😁 

Kidding aside: agreed, i usually just stick with the double indent for continuation rather than try lining up things :-)